sstables/checksummed_data_source.hh

```diff
@@ -15,15 +15,19 @@
 
 namespace sstables {
 
+using integrity_error_handler_t = std::function<void(seastar::sstring)>;
```

Comment:
`integrity_error_handler` is a good enough name, the `_t` suffix is not needed (we don't really use it).

Reply:
You should also declare here (and define in the `.cc` file) a default error handler:

```
integrity_error_handler throwing_integrity_error_handler;
```

So that users have a ready-to-use one for the default case. Maybe even make this the default argument for `error_handler`. We usually don't like default arguments but in this case we have a default used by almost all the callers and we have one exceptional user who will use a different one.

Reply:
> `integrity_error_handler` is a good enough name, the `_t` suffix is not needed (we don't really use it).

Ack, we do use it in some places, but in general we are not consistent.
```
$ git grep 'using [a-zA-Z0-9_]* = std::function' | wc -l
78
$ git grep 'using [a-zA-Z0-9_]*_t = std::function' | wc -l
12
$ git grep 'using [a-zA-Z0-9_]*_type = std::function' | wc -l
14
$ git grep 'using [a-zA-Z0-9_]*_func = std::function' | wc -l
14
$ git grep 'using [a-zA-Z0-9_]*_fn = std::function' | wc -l
11
```